FuncGodot Manual/pages/tips_runtime_building.html

```diff
@@ -0,0 +1,134 @@
+# in order to add more functionality after building.
+func _ready():
+    func_godot_map.connect("build_complete", _build_complete)
+    func_godot_map.connect("build_failed", _build_failed)
+    func_godot_map.connect("unwrap_uv2_complete", _unwrap_uv2_complete)
+    func_godot_map.local_map_file = "res://maps/runtime_building.map"
+    func_godot_map.verify_and_build()
+
+# We need to wait for the build to finish before unwrapping mesh UV2s for lightmap baking.
+# Currently lightmap baking cannot be performed at runtime, so it's not terribly useful yet.
+# Consider creating a procedural Voxel GI solid class entity that can be baked at runtime instead.
+func _build_complete() -> void:
+    print("Success! Unwrapping UV2...")
+    func_godot_map.unwrap_uv2()
+
+func _build_failed() -> void:
+    printerr("Failed to build the map file! :(")
+
+func _unwrap_uv2_complete() -> void:
+    print("UV2 unwrapping completed!")</code></pre>
+    <p>An example of this can be found in <a href="https://github.com/func-godot/func_godot_runtime_building" target="_blank">func_godot_runtime_building repository</a>.</p>
+
+    <h1>Building from Map Editors</h1>
+    <p>
+        Much like idTech engine games, it's possible to build and launch directly into a map from NetRadiant Custom and TrenchBroom alike.
+        This can be quite a helpful tool for mappers to quickly try out their creations without a lengthy startup or debug menu process.
+    </p>
+
+    <h2>Godot project setup example</h2>
+
+    <p>
+        Regardless of which map editor you use, you'll need to check for any 
+        <a href="https://docs.godotengine.org/en/stable/classes/class_os.html#class-os-method-get-cmdline-user-args">command line arguments.</a>
+
+        <p align="center"><img src="../images/tips_editorbuild_1.png"><br></p>
+
+        A basic example might look something like this:
+
+    </p>
+
+    <code><pre>
+# Somewhere in a script called when your project opens. Make sure you have a 
+# valid reference to a FuncGodot map or create one and apply your map settings to it.
+
+var func_godot_map: FuncGodotMap
+
+var map_override: String
+var cmd_line_args := OS.get_cmdline_user_args()
+
+for arg in cmd_line_args:
+    # Example argument, with a known delimiter.
+    if arg.contains("launch_debug::"):
+        var split := arg.split("::")
+
+        # Basic input validation
+        if split.size != 2:
+        return
+
+        map_override = arg.split("::")[1]
+
+        func_godot_map.local_map_file = map_override
+        func_godot_map.verify_and_build()</code></pre>
+    <h2>Editor setup</h2>
+    <h3>TrenchBroom</h3>
+    <p>
+        The TrenchBroom documenation <a href="https://trenchbroom.github.io/manual/latest/#launching_game_engines">covers this</a>. You configure an engine you can run from which is its executable, and you pass a parameter to it, which is the map file. Trenchbroom provides variables for finding the path of the map file.
```

Comment:
Strange phrasing and not thorough enough of a tutorial on it. Should rethink the whole thing.

Try setting your project.godot file to open up with Notepad or some other text editor instead (at least temporarily) in order to test out a bare minimum project to launch a map from the Godot editor. Also use the default FuncGodot game configuration to perform the launch from TrenchBroom.

I'd recommend doing this for NetRadiant Custom as well. New screenshots are in order anyway.

Reply:
> Try setting your project.godot file to open up with Notepad or some other text editor instead (at least temporarily) in order to test out a bare minimum project to launch a map from the Godot editor. Also use the default FuncGodot game configuration to perform the launch from TrenchBroom.

I don't follow what you're asking here? Launching a godot project from notepad? I can detail steps for doing this including screenshots, but all this is, is adding a configuration for the engine which tells it where the binary is, and passing a CLI parameter for where the .map file is. It is an extremely short process, and a lot of the extensive detail is in the trenchbroom documentation which I don't think we should be duplicating.

Regarding NetRadiant, I don't use it, so can't comment on that or add screenshots for it. @sinewavey any thoughts?

Reply:
I will remake NRC screenshots

Reply:
> I don't follow what you're asking here? Launching a godot project from notepad?

I guess what I mean is basically don't use your project.godot file as your launch executable (for the tutorial, not your own personal flow), since by default this is not how project.godot files open.

It may be an extremely short process, but it still has multiple steps and logic to it.

- Set your Godot Exe as the engine to launch from.
- Pass the command line argument to point to your project.
- Pass the command line argument to launch your map.

We may need to just rethink how this section is written altogether tbh, especially since the above 3 steps will be common to any map editor. In that case it would make the TrenchBroom explainer be as simple as linking the docs and not even needing its own section at all.

I'd have to reread through the NetRadiant Custom section, but it seems like it's pretty much the same? We might just be overcomplicating this whole Editor section.

Reply:
Between the two it basically the same, just varying in syntax. The important common thread is the Godot command line arguments part

Reply:
I've added TrenchBroom tutorial for launching from TrenchBroom. Happy to hear other thoughts